### PR TITLE
(FACT-1611) update external fact stderr test

### PR DESCRIPTION
**Pull request by ericldelaney**
I moved the external fact that tests that we only run a test once
I added a test to verifyt that stderr message from external facts show up in the stderr from facter o puppet facts
Updated for code style, TestRail ID, risk tag

### CODE REVIEW

**Comment by ericldelaney** [comment]
Tested on
arista4-32a, ciscoxr-64a%7Bhypervisor=vmpooler%7D, cisconx-64a%7Bhypervisor=vmpooler%7D, cumulus25-64a, debian8-64a, fedora25-64a, osx1012-64a, redhat7-64a, sles12-64a, solaris11-64a, ubuntu1604-64a, windows10ent-64a, windows2016-64a

File: acceptance/tests/external_facts/external_facts_only_run_once.rb [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+        if times == 1
+          step "External executable fact executed once"
+        else
+          fail_test "External fact executed #{times} times, expected once: #{facter_output.stderr}"
+        end
+      end
+    end
+  end
+end
+
```
Comment (johnduarte) [inline acceptance/tests/external_facts/external_facts_only_run_once.rb]: This blank line should be removed.

File: acceptance/tests/external_facts/external_fact_stderr_messages_output_to_stderr.rb [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+
+    step "Agent #{agent}: external fact stderr messages should appear on stderr from puppet facts" do
+      on(agent, puppet("facts")) do |puppet_output|
+        assert_match(/Warning.*SCRIPT STDERR/, puppet_output.stderr,
+                     "Expected puppet facts to output a warning message with the stderr string from the external fact")
+      end
+    end
+  end
+end
+
```
Comment (johnduarte) [inline acceptance/tests/external_facts/external_fact_stderr_messages_output_to_stderr.rb]: This blank line should be removed.

File: acceptance/tests/external_facts/external_facts_only_run_once.rb [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+      on(agent, "mkdir -p '#{factsd}'")
+      create_remote_file(agent, ext_fact, content)
+      on(agent, "chmod +x '#{ext_fact}'")
+    end
+
+    step "Agent #{agent}: ensure the fact is only executed once" do
+      on(agent, facter) do |facter_output|
+        lines = facter_output.stderr.split('\n')
+        times = lines.count { |line| line =~ /SCRIPT CALLED/ }
+        if times == 1
```
Comment (johnduarte) [inline acceptance/tests/external_facts/external_facts_only_run_once.rb]: This seems verbose. Suggested edit
```
assert_equal(1, times, "External fact executed #{times} times, expected once: #{facter_output.stderr}")
```

**Comment by ericldelaney** [comment]
@johnduarte Updated with your comments

**Comment by johnduarte** [comment]
Out of ignorant curiosity, why is this targeted at stable vs master?

**Comment by ericldelaney** [comment]
That's where I was told to target them. After a while Maggie normally rolls everything up and merges them to master.

**Pull request merged**